apps/shell/app.js

```diff
@@ -52,17 +55,38 @@ app.set('views', path.join(__dirname, 'views'));
 app.use(process.env.PASSENGER_BASE_URI || '/', router);
 
 // Setup websocket server
-var server = new http.createServer(app);
-var wss = new WebSocket.Server({ noServer: true });
+const server = new http.createServer(app);
+const wss = new WebSocket.Server({ noServer: true });
+
+let host_whitelist = [];
+if (process.env.SSHHOST_WHITELIST){
+  host_whitelist = process.env.SSHHOST_WHITELIST.split(':');
+}
+
+glob.sync(path.join((process.env.OOD_CLUSTERS || '/etc/ood/config/clusters.d'), '*.y*ml'))
+  .map(yml => yaml.safeLoad(fs.readFileSync(yml)))
+  .filter(config => (config.v2 && config.v2.login && config.v2.login.host) && ! (config.v2 && config.v2.metadata && config.v2.metadata.hidden))
+  .forEach((config) => {
+    let host = config.v2.login.host; //Already did checking above
+    let isDefault = config.v2.login.default;
+    if(isDefault){
+      host_whitelist.unshift(host);
+    } else{
+      host_whitelist.push(host);
+    }
+  });
+host_whitelist.push('default');
+
+host_whitelist = Array.from(new Set(host_whitelist)); // remove duplicates
```

Comment:
If you start with host_whitelist as a Set and use a separate `let default_sshhost;` before the start of the `glob.sync` block, you could just update `default_sshhost` value in that block whenever a cluster config species `config.v2.login.default` and then the order of the whitelist is irrelevant.